### PR TITLE
feat(route): add comicat 漫喵动漫BT下载

### PR DESCRIPTION
**Pull request by Cyang39**
## 该 PR 相关 Issue / Involved Issue

Close #

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/comicat/search/动漫国字幕组+跃动青春+720P
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
通过漫喵动漫的搜索功能订阅动漫新番的更新
站点所有动漫的更新有官方的 seed，所以不需要

### CODE REVIEW

File: lib/v2/comicat/maintainer.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    '/issue/:user/:repo?': ['Cyang39'],
```
Comment (TonyRL) [inline lib/v2/comicat/maintainer.js:2]: https://github.com/DIYgod/RSSHub/blob/21a30a5cf3c461ea0953085994c55ca8f8bfa83f/lib/v2/comicat/router.js#L2
```suggestion
    '/search/:keyword': ['Cyang39'],
```

File: docs/anime.md [hunk 2]
```diff
@@ -191,6 +191,12 @@ pageClass: routes
 
 <Route author="TonyRL" example="/creative-comic/book/117" path="/creative-comic/book/:id/:coverOnly?/:quality?" :paramsDesc="['漫畫 ID，可在 URL 中找到', '僅獲取封面，非 `true` 時將獲取**全部**頁面，預設 `true`', '閱讀品質，標準畫質 `1`，高畫質 `2`，預設 `1`']" radar="1" rssbud="1"/>
 
+## Comicat
+
+### 搜索关键词
+
+<Route author="Cyang39" example="/comicat/search/喵萌奶茶屋+跃动青春+720P+简体" path="/comicat/search/:keyworkd" :paramsDesc="['关键词，请用`+`号连接']" supportBT=1/>
```
Comment (TonyRL) [inline docs/anime.md:198]: https://github.com/DIYgod/RSSHub/blob/21a30a5cf3c461ea0953085994c55ca8f8bfa83f/lib/v2/comicat/router.js#L2
```suggestion
<Route author="Cyang39" example="/comicat/search/喵萌奶茶屋+跃动青春+720P+简体" path="/comicat/search/:keyword" :paramsDesc="['关键词，请用`+`号连接']" supportBT=1/>
```

The example also returns nothing.

File: lib/v2/comicat/search.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+
+const baseUrl = 'https://comicat.org';
+
+module.exports = async (ctx) => {
+    const keyword = ctx.params.keyword;
+    const { data: response } = await got(`${baseUrl}//search.php?keyword=${encodeURIComponent(keyword)}}`);
```
Comment (TonyRL) [inline lib/v2/comicat/search.js:9]:
```suggestion
    const { data: response } = await got(`${baseUrl}/search.php?keyword=${encodeURIComponent(keyword)}}`);
```

File: lib/v2/comicat/search.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                author: $(item).find('td:nth-child(8)').text().trim(),
+            }));
+
+    const items = await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const { data: response } = await got(item.link);
+                const $ = cheerio.load(response);
+
+                item.description = `magnet:?xt=urn:btih:${$('#text_hash_id').text().split('，特征码：')[1]}`.trim();
```
Comment (TonyRL) [inline lib/v2/comicat/search.js:27]: Adding a magnet link in `description` won't make the RSS feed supported by BT client.
docs: https://docs.rsshub.app/joinus/advanced-feed.html

Comment (Cyang39) [reply lib/v2/comicat/search.js]: thks, I put it in `enclosure_url` field, and add the page main content as `description`

File: docs/anime.md [hunk 5]
```diff
@@ -191,6 +191,12 @@ pageClass: routes
 
 <Route author="TonyRL" example="/creative-comic/book/117" path="/creative-comic/book/:id/:coverOnly?/:quality?" :paramsDesc="['漫畫 ID，可在 URL 中找到', '僅獲取封面，非 `true` 時將獲取**全部**頁面，預設 `true`', '閱讀品質，標準畫質 `1`，高畫質 `2`，預設 `1`']" radar="1" rssbud="1"/>
 
+## Comicat
+
+### 搜索关键词
+
+<Route author="Cyang39" example="/comicat/search/喵萌奶茶屋+跃动青春+720P+简体" path="/comicat/search/:keyword" :paramsDesc="['关键词，请用`+`号连接']" supportBT=1/>
```
Comment (TonyRL) [inline docs/anime.md:198]: The example https://www.comicat.org/search.php?keyword=喵萌奶茶屋+跃动青春+720P+简体 returns nothing.

File: lib/v2/comicat/search.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+module.exports = async (ctx) => {
+    const keyword = ctx.params.keyword;
+    const { data: response } = await got(`${baseUrl}/search.php?keyword=${encodeURIComponent(keyword)}}`);
+    const $ = cheerio.load(response);
+    const list = $('#listTable tbody > tr')
+        .toArray()
+        .map((item) => ({
+                title: $(item).find('td:nth-child(3)').text().trim(),
+                link: `${baseUrl}/${$(item).find('td:nth-child(3) a').attr('href')}`,
+                pubDate: parseDate($(item).find('td:nth-child(1)').text().trim()),
```
Comment (TonyRL) [inline lib/v2/comicat/search.js:16]: Parse the pubDate at item level instead. The first two pubDate in automated test are invalid.

File: docs/anime.md (hunk 5 above)
Comment (Cyang39) [reply docs/anime.md]: Sorry, I not cheked the example well, it should be https://www.comicat.org/search.php?keyword=喵萌奶茶屋+跃动青春+720P+简日, i will change it in next commit.

File: lib/v2/comicat/search.js (hunk 6 above)
Comment (Cyang39) [reply lib/v2/comicat/search.js]: When I move pubDate into item level, the invalid problem seems gone.

File: lib/v2/comicat/search.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+
+const baseUrl = 'https://comicat.org';
+
+module.exports = async (ctx) => {
+    const keyword = ctx.params.keyword;
+    const { data: response } = await got(`${baseUrl}/search.php?keyword=${encodeURIComponent(keyword)}}`);
```
Comment (TonyRL) [inline lib/v2/comicat/search.js:9]:
```suggestion
    const { data: response } = await got(`${baseUrl}/search.php?keyword=${encodeURIComponent(keyword)}`);
```

**Pull request merged**